.editorconfig

```diff
@@ -1,36 +1,59 @@
 # EditorConfig configuration for nixpkgs
-# http://EditorConfig.org
+# https://EditorConfig.org
 
 # Top-most EditorConfig file
 root = true
 
-# Unix-style newlines with a newline ending every file, utf-8 charset
+# utf-8 charset, unix-style newlines, indent width of two with spaces, newline ending, trim whitespace
 [*]
+charset = utf-8
 end_of_line = lf
+indent_size = 2
+indent_style = space
 insert_final_newline = true
 trim_trailing_whitespace = true
-charset = utf-8
 
 # Ignore diffs/patches
 [*.{diff,patch}]
 end_of_line = unset
+indent_size = unset
+indent_style = unset
 insert_final_newline = unset
 trim_trailing_whitespace = unset
 
 # see https://nixos.org/nixpkgs/manual/#chap-conventions
 
-# Match nix/perl/python/ruby/shell/docbook files, set indent to spaces
-[*.{nix,pl,py,rb,sh,xml}]
-indent_style = space
-
 # Match docbook files, set indent width of one
 [*.xml]
 indent_size = 1
 
-# Match nix/ruby files, set indent width of two
-[*.{nix,rb}]
-indent_size = 2
-
 # Match perl/python/shell scripts, set indent width of four
-[*.{pl,py,sh}]
+[*.{pl,pm,py,sh}]
 indent_size = 4
+
+# Disable for c/conf/in files
+[*.{c,conf,in}]
+indent_size = unset
+indent_style = unset
+
+# Disable for auto-generated gemset.nix files
+[gemset.nix]
+insert_final_newline = unset
+
+# Disable for auto-generated melpa file
+[pkgs/applications/editors/emacs-modes/recipes-archive-melpa.json]
+indent_size = unset
+
+# Disable for auto-generated hackage2nix file
+[pkgs/development/haskell-modules/hackage-packages.nix]
+indent_style = unset
+indent_size = unset
+trim_trailing_whitespace = unset
+
+# Disable for auto-generated node-packages files
+[pkgs/development/node-packages/node-packages-v*.nix]
+insert_final_newline = unset
+
+# Disable for auto-generated perl-packages file
+[pkgs/top-level/perl-packages.nix]
```

Comment:
Some of the stuff might be automatic. Some of it is manual. But there might be a lot of errors still though given its size.